File: backend/app/schemas/lead.py
from pydantic import BaseModel, EmailStr, Field, HttpUrl
from typing import Optional, Dict, Any, List
from datetime import datetime


class LeadBase(BaseModel):
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    email: Optional[EmailStr] = None
    title: Optional[str] = None
    company: Optional[str] = None
    company_domain: Optional[str] = None
    company_size: Optional[str] = None
    industry: Optional[str] = None
    location: Optional[str] = None
    phone: Optional[str] = None
    linkedin_url: Optional[str] = None
    twitter_url: Optional[str] = None
    website_url: Optional[str] = None
    notes: Optional[str] = None
    source: Optional[str] = None


class LeadCreate(LeadBase):
    project_id: int


class LeadUpdate(LeadBase):
    status: Optional[str] = None
    score: Optional[float] = None
    enrichment_data: Optional[Dict[str, Any]] = None


class LeadSearch(BaseModel):
    industry: Optional[str] = None
    title: Optional[str] = None
    location: Optional[str] = None
    company_size: Optional[str] = None
    keywords: Optional[str] = None
    project_id: Optional[int] = None
    min_score: Optional[float] = None
    status: Optional[str] = None
    limit: int = 50
    offset: int = 0


class Lead(LeadBase):
    id: int
    status: str
    score: float
    project_id: int
    owner_id: int
    enrichment_data: Optional[Dict[str, Any]] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        orm_mode = True


class LeadWithScore(Lead):
    score_explanation: Optional[Dict[str, Any]] = None

File: shared/types/lead.ts
/**
 * Shared type definitions for leads
 */

export interface Lead {
  id: number;
  first_name: string;
  last_name: string;
  email?: string;
  predicted_email?: string;
  email_confidence?: number;
  title?: string;
  company: string;
  company_domain?: string;
  company_size?: string;
  industry?: string;
  location?: string;
  phone?: string;
  linkedin_url?: string;
  twitter_url?: string;
  website_url?: string;
  source?: string;
  notes?: string;
  status: string;
  score: number;
  enrichment_data?: any;
  created_at: string;
  updated_at?: string;
  project_id: number;
  owner_id: number;
}

export interface LeadSearchParams {
  industry?: string;
  title?: string;
  location?: string;
  company_size?: string;
  keywords?: string;
  project_id?: number;
  min_score?: number;
  status?: string;
  limit?: number;
  offset?: number;
}

export interface LeadCreate {
  first_name: string;
  last_name: string;
  email?: string;
  title?: string;
  company: string;
  company_domain?: string;
  company_size?: string;
  industry?: string;
  location?: string;
  linkedin_url?: string;
  source?: string;
  notes?: string;
  project_id: number;
}

export interface LeadUpdate {
  first_name?: string;
  last_name?: string;
  email?: string;
  title?: string;
  company?: string;
  company_domain?: string;
  company_size?: string;
  industry?: string;
  location?: string;
  phone?: string;
  linkedin_url?: string;
  twitter_url?: string;
  website_url?: string;
  source?: string;
  notes?: string;
  status?: string;
  score?: number;
  enrichment_data?: any;
}

export interface LeadEnrichRequest {
  lead_id: number;
}

export interface LeadBatchEnrichRequest {
  lead_ids: number[];
  project_id?: number;
}

export interface LeadScoreExplanation {
  score: number;
  reasons: string[];
  factors: {
    [key: string]: {
      value: any;
      importance: number;
    }
  };
}

export interface LeadWithScore extends Lead {
  score_explanation?: LeadScoreExplanation;
}
